src/Common/Common.service.ts

```diff
@@ -169,19 +200,107 @@ const cdMaterialListModal = (artifacts: any[], offset: number, artifactId?: numb
+        .filter((user) => user.startsWith(API_TOKEN_PREFIX))
+        .sort(stringComparatorBySortOrder)
+    const specificUsersEmails = userApprovalConfig.specificUsers.identifiers
+        .filter((user) => !user.startsWith(API_TOKEN_PREFIX) && user !== DefaultUserKey.system)
+        .sort(stringComparatorBySortOrder)
+
+    const specificUsersData: ImageApprovalPolicyType['specificUsersData'] = {
+        dataStore: userApprovalConfig.specificUsers.identifiers.reduce(
+            (acc, email) => {
+                acc[email] = {
+                    email,
+                    hasAccess: approvalUsersMap[email] ?? false,
+                }
+                return acc
+            },
+            {} as Record<string, UserGroupApproverType>,
+        ),
+        requiredCount: userApprovalConfig.specificUsers.requiredCount,
+        emails: specificUsersEmails.concat(specificUsersAPIToken),
+    }
+
+    const validGroups = userApprovalConfig.userGroups.map((group) => group.identifier)
+
+    const usersList = Object.keys(imageApprovalUsersInfo).filter((user) => user !== DefaultUserKey.system)
+    const groupIdentifierToUsersMap = usersList.reduce(
+        (acc, user) => {
+            const userGroups = imageApprovalUsersInfo[user] || []
+            userGroups.forEach((group) => {
+                if (!acc[group.identifier]) {
+                    acc[group.identifier] = {}
+                }
+                acc[group.identifier][user] = true
+            })
+            return acc
+        },
+        {} as Record<string, Record<string, true>>,
+    )
+
+    return {
+        isPolicyConfigured,
+        specificUsersData,
+        userGroupData: userApprovalConfig.userGroups.reduce(
+            (acc, group) => {
+                const identifier = group.identifier
+                // No need of handling api tokens here since they are not part of user groups
+                const users = Object.keys(groupIdentifierToUsersMap[identifier] || {}).sort(stringComparatorBySortOrder)
+
+                acc[identifier] = {
+                    dataStore: users.reduce(
+                        (acc, user) => {
+                            acc[user] = {
+                                email: user,
+                                hasAccess: approvalUsersMap[user] ?? false,
+                            }
+                            return acc
+                        },
+                        {} as Record<string, UserGroupApproverType>,
+                    ),
+                    requiredCount: group.requiredCount,
+                    emails: users,
+                }
+
+                return acc
+            },
+            {} as Record<string, ImageApprovalPolicyUserGroupDataType>,
+        ),
+        // Not sorting since would change them in approval info modal to name
+        validGroups,
+    }
+}
+
 const processCDMaterialsApprovalInfo = (enableApproval: boolean, cdMaterialsResult): CDMaterialsApprovalInfo => {
```

Comment:
All these methods to compute data for cd material should belong to utils can take it as task or will do that later after testing